### PR TITLE
test(BDD): add initial BDD for node drain

### PR DESCRIPTION
**Pull request by paulyoong**
Describe the initial behaviours of the node drain feature.

### CODE REVIEW

File: tests/bdd/features/drain/node/feature.feature [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Feature: Drain
+
+  Background:
+    Given Multiple uncordoned nodes
+
+  Scenario: Draining a node
+    Given an uncordoned node
+    When the user issues a drain command to the node
+    Then the node should be cordoned
+    And nexuses should be moved to other suitable nodes
```
Comment (tiagolobocastro) [inline tests/bdd/features/drain/node/feature.feature:10]: nit: "its nexuses"?

File: tests/bdd/features/drain/node/feature.feature [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    When the user issues a drain command to the node
+    Then the node should remain cordoned
+    And nexuses should be moved to other suitable nodes
+
+# The system is unable to move all resources from the draining node to another node
+  Scenario: Draining a node with insufficient system capacity
+    Given an uncordoned node
+    And insufficient system-wide capacity on other nodes to accommodate all drained resources
+    When the user issues a drain command to the node
+    Then the drain should fail to complete in its entirety
```
Comment (tiagolobocastro) [inline tests/bdd/features/drain/node/feature.feature:23]: Do we need a use case for the system resources changing during the drain? Eg: initially we have enough capacity, but as we're draining, that changes for some reason (example: more volumes were created, or a node is lost, or even cordoned)

**Review by tiagolobocastro — APPROVED** [review]

File: tests/bdd/features/drain/node/feature.feature (hunk 1 above)
Comment (paulyoong) [reply tests/bdd/features/drain/node/feature.feature]: Done.

File: tests/bdd/features/drain/node/feature.feature (hunk 2 above)
Comment (paulyoong) [reply tests/bdd/features/drain/node/feature.feature:23]: Yes good idea. I have added an additional scenario. Can you take a look and let me know what you think?

**Comment by paulyoong** [comment]
bors merge

**Pull request merged**